examples/app-secrets.yaml.tmpl

```diff
@@ -0,0 +1,15 @@
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: app-secrets
+spec:
+  provider: gcp
```

Comment:
How does the value of `gcp` here indicate which binary to execute in the CSI driver? Is it based on the name of the subdirectory within the host path?

Reply:
Its not really defined anywhere but that the provider name should have a binary at:

`<plugin_dir>/<provider>/provider-<provider>`

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/3f60fa0a71bb848878c372675be91d9d2c1d389b/pkg/secrets-store/utils.go#L68-L74

I tried to make this a bit more explicit by adding 

```
const providerName = "gcp"
```